tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/globalvar/GlobalVarNegativeTest.java

```diff
@@ -33,12 +33,38 @@
     public void testGlobalVarNegatives() {
         CompileResult resultNegative = BCompileUtil.compile(
                 "test-src/statements/variabledef/global_variable_negative.bal");
-        Assert.assertEquals(resultNegative.getErrorCount(), 6);
+        Assert.assertEquals(resultNegative.getErrorCount(), 4);
         BAssertUtil.validateError(resultNegative, 0, "invalid token 'int'", 27, 8);
         BAssertUtil.validateError(resultNegative, 1, "invalid token 'int'", 29, 8);
-        BAssertUtil.validateError(resultNegative, 2, "mismatched input ';'. expecting '='", 31, 32);
-        BAssertUtil.validateError(resultNegative, 3, "mismatched input ';'. expecting '='", 33, 27);
-        BAssertUtil.validateError(resultNegative, 4, "mismatched input ';'. expecting '='", 35, 46);
-        BAssertUtil.validateError(resultNegative, 5, "mismatched input ';'. expecting '='", 37, 59);
+        BAssertUtil.validateError(resultNegative, 2, "mismatched input ';'. expecting '='", 31, 46);
+        BAssertUtil.validateError(resultNegative, 3, "mismatched input ';'. expecting '='", 33, 59);
+    }
+
+    @Test
+    void testGlobalVariableInitNegative() {
+        CompileResult result = BCompileUtil.compile("test-src/statements/variabledef/global_variable_init_negative" +
+                ".bal");
+
+        Assert.assertEquals(result.getErrorCount(), 8);
+        int i = 0;
+        BAssertUtil.validateError(result, i++, "uninitialized variable 'i'", 17, 1);
+        BAssertUtil.validateError(result, i++, "uninitialized variable 's'", 18, 1);
+        BAssertUtil.validateError(result, i++, "uninitialized variable 'a'", 19, 1);
+        BAssertUtil.validateError(result, i++, "uninitialized variable 'b'", 20, 1);
+        BAssertUtil.validateError(result, i++, "variable 'i' is not initialized", 25, 5);
+        BAssertUtil.validateError(result, i++, "variable 'i' is not initialized", 31, 5);
+        BAssertUtil.validateError(result, i++, "variable 'a' is not initialized", 39, 13);
+        BAssertUtil.validateError(result, i, "variable 's' is not initialized", 40, 18);
+    }
+
+    @Test
+    void testGlobalVariableInitWithInvocationNegative() {
+        CompileResult result = BCompileUtil.compile("test-src/statements/variabledef" +
```

Comment:
This test-case is failing intermittently. Shall I disable it for now? Can you create an issue for this?

Reply:
created https://github.com/ballerina-platform/ballerina-lang/issues/23211 to track this.